### PR TITLE
update PIConGPU recipe

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- update requirements for the current develop
- increase ISAAC requirements for 0.5.0 and develop

This is the first step to update the PIConGPU container to Ubuntu 20.4. 
I have not updated the supported compiler configurations.

### CODE REVIEW

File: packages/picongpu/package.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 #
-# Authors: Axel Huebl
+# Authors: Axel Huebl, Rene Widera
```
Comment (ax3l) [inline packages/picongpu/package.py:6]: Fantastic, can you also add yourself to the `maintainers` member list please? :)

File: packages/picongpu/package.py [hunk 2]
```diff
@@ -84,8 +85,9 @@ class Picongpu(Package):
     depends_on('adios@1.13.1:,develop', when='+adios')
     depends_on('isaac@1.4.0', when='@:0.4.3 +isaac')
     depends_on('isaac-server@1.4.0', type='run', when='@:0.4.3 +isaac')
-    depends_on('isaac@1.5.0,develop', when='@0.5.0: +isaac')
-    depends_on('isaac-server@1.5.0,develop', type='run', when='@0.5.0: +isaac')
+    depends_on('isaac@1.5.2,develop', when='@0.5.0: +isaac')
+    depends_on('isaac-server@1.5.2,develop', type='run', when='@0.5.0: +isaac')
+    depends_on('openpmd-api@dev', type='run', when='@develop')
```
Comment (ax3l) [inline packages/picongpu/package.py]: This is a regular lib dependency:
```suggestion
    depends_on('openpmd-api@dev', when='@develop')
```

You can also add a variant for it as we have for HDF5, so you can control the dependencies it pulls a bit better:
```py
    variant('openpmd', default=True, description='Enable openPMD I/O')
```
Then this line will read:
```suggestion
    depends_on('openpmd-api@dev', when='@develop +openPMD')
```

File: packages/picongpu/package.py (hunk 1 above)
Comment (psychocoderHPC) [reply packages/picongpu/package.py:6]: added me to the maintainer list

File: packages/picongpu/package.py (hunk 2 above)
Comment (psychocoderHPC) [reply packages/picongpu/package.py]: fixed

**Comment by psychocoderHPC** [comment]
@ax3l could you please merge this PR. I forgot that this is open but integrated the your suggestions now.

**Review by ax3l — APPROVED** [review]
LGTM, thanks!

File: packages/picongpu/package.py [hunk 3]
```diff
@@ -84,8 +87,9 @@ class Picongpu(Package):
     depends_on('adios@1.13.1:,develop', when='+adios')
     depends_on('isaac@1.4.0', when='@:0.4.3 +isaac')
     depends_on('isaac-server@1.4.0', type='run', when='@:0.4.3 +isaac')
-    depends_on('isaac@1.5.0,develop', when='@0.5.0: +isaac')
-    depends_on('isaac-server@1.5.0,develop', type='run', when='@0.5.0: +isaac')
+    depends_on('isaac@1.5.2,develop', when='@0.5.0: +isaac')
+    depends_on('isaac-server@1.5.2,develop', type='run', when='@0.5.0: +isaac')
+    depends_on('openpmd-api@0.13.2:,dev', when='@develop' +openpmd)
```
Comment (ax3l) [inline packages/picongpu/package.py]:
```suggestion
    depends_on('openpmd-api@0.13.2:,dev', when='@develop +openpmd')
```

**Review by ax3l — CHANGES_REQUESTED** [review]
Uff, one more bug to squash

**Review by ax3l — APPROVED** [review]

File: packages/picongpu/package.py (hunk 3 above)
Comment (psychocoderHPC) [reply packages/picongpu/package.py]: thx for fixing

**Pull request merged**